samtranslator/model/apigateway.py

```diff
@@ -395,8 +395,15 @@ def _get_type(self):
         return "LAMBDA"
 
     def _get_identity_header(self):
-        if not self.identity or not self.identity.get("Header"):
-            return "Authorization"
+        try:
+            if not self.identity or not self.identity.get("Header"):
+                return "Authorization"
+        except AttributeError:
+            raise InvalidResourceException(
+                self.api_logical_id,
+                "Auth.Authorizers.<Authorizer>.Identity must be a dict (LambdaTokenAuthorizationIdentity "
+                "or LambdaRequestAuthorizationIdentity).",
+            )
```

Comment:
Instead of `try/except` block, shouldn't we check if the property is an expected type (dict in this case)?